### ISSUE TITLE
Inefficient API is used

### ISSUE THREAD

**Issue by struggggle** [TeamNewPipe/NewPipe#459]
Dear developers,
I found an inefficient API, setImageResource(), is widely used in NewPipe.

For setImageResource(), Google says this:
https://developer.android.com/reference/android/widget/ImageView.html
" This does Bitmap reading and decoding on the UI thread, which can cause a latency hiccup. If that's a concern, consider using setImageDrawable(android.graphics.drawable.Drawable) or setImageBitmap(android.graphics.Bitmap) and BitmapFactory instead. "

I noticed that in the following code, setImageResource() is used:
(1) org.schabi.newpipe.info_list.InfoItemBuilder.java    (line number: 154, 176)
https://github.com/TeamNewPipe/NewPipe/blob/master/app/src/main/java/org/schabi/newpipe/info_list/InfoItemBuilder.java

(2) org.schabi.newpipe.detail.VideoItemDetailFragment.java   (line number: 187, 190, 337)
https://github.com/TeamNewPipe/NewPipe/blob/master/app/src/main/java/org/schabi/newpipe/detail/VideoItemDetailFragment.java

Do you think the use of setImageResource() will affect NewPipe's performance? If yes, maybe we should replace it with setImageDrawable() or setImageBitmap().

Thanks.

**Comment by KeizerDev** [comment]
@struggggle Why not do this? Better performance or not, if the documenation says it's inefficient we should change!

**Comment by ranomier** [comment]
+1

**Comment by SkyzohKey** [comment]
@theScrabi Just mentioning you as it seem you didn't saw (or saw and didn't responded, idk ^^) that issue. Performances is something wanted if you want Newpipe to win ! :1st_place_medal:

**Comment by mauriciocolli** [comment]
Well just replacing the call won't solve anything, because it to call them, you've to have an already loaded `Bitmap` or `Drawable`, thus, you have to implement some sort of background thread to load it and then call it.

We already use an ImageLoader to load most of the images, these calls are just for the initialization of these views, and in fact, just inflating the layout is a more expensive call than this.

I'm planning in change the current ImageLoader (some reasons here #545), so then we can use it to show every image, either by network or local images, solving this issue (I'm testing image loaders yet).

**Comment by struggggle** [comment]
Yes, the key is to move Bitmap reading and decoding to a background thread.

**Comment by SkyzohKey** [comment]
Up 👆 so it don't closes.

**Comment by opusforlife2** [comment]
Is this still the case today?

**Comment by mhmdanas** [comment]
@opusforlife2 yes, this still appears to be relevant. `setImageResource`'s documentation regarding the "latency hiccups" hasn't changed, and I found 16 occurrences of `setImageResource` in NewPipe.

**Comment by PotatoMasherGUI** [comment]
Devs could follow this steps:

    1. Replace all instances of imageView.setImageResource(resourceId) with imageView.setImageDrawable(ContextCompat.getDrawable(context, resourceId)). This will load the drawable on a background thread instead of the UI thread, reducing latency.

    2. Alternatively, replace imageView.setImageResource(resourceId) with BitmapFactory.decodeResource(context.getResources(), resourceId) to get a Bitmap object from the resource and then set it using imageView.setImageBitmap(bitmap). This also avoids decoding the bitmap on the UI thread.

By making these changes, NewPipe's performance will be improved and the potential latency hiccups caused by using setImageResource() will be avoided.
